File: packages/swift/Sources/Blasphem/Judgement.swift
/// One verdict for one message.
public struct Judgement: Equatable, Sendable {
    /// True when no nudge is due. Unroutable text is safe; the nudge fails open.
    public let safe: Bool
    /// Ordinal risk from 0 through 1. Not a probability.
    public let score: Double
    /// The locale that produced the score, or nil.
    public let locale: String?
    /// The masked text when `grawlix` was requested, otherwise nil.
    public let grawlix: String?

    public init(safe: Bool, score: Double, locale: String?, grawlix: String?) {
        self.safe = safe
        self.score = score
        self.locale = locale
        self.grawlix = grawlix
    }
}

File: packages/android/engine/src/main/kotlin/me/sospedra/blasphem/Judgement.kt
package me.sospedra.blasphem

/** One verdict for one message. */
data class Judgement(
    /** True when no nudge is due. Unroutable text is safe; the nudge fails open. */
    val safe: Boolean,
    /** Ordinal risk from 0 through 1. Not a probability. */
    val score: Double,
    /** The locale that produced the score, or null. */
    val locale: String?,
    /** Masked text for unsafe verdicts when requested, otherwise null. */
    val grawlix: String?,
)
